saleor/graphql/core/connection.py

```diff
@@ -1,11 +1,227 @@
-from typing import Union
+from typing import Any, Dict, List, Tuple, Union
 
 import graphene
 import opentracing as ot
-from django.db.models import Manager, QuerySet
-from graphene import Field, List, NonNull, ObjectType, String
+from django.db.models import Manager, Model as DjangoModel, Q, QuerySet
 from graphene.relay.connection import Connection
 from graphene_django_optimizer.types import OptimizedDjangoObjectType
+from graphql_relay.connection.connectiontypes import Edge, PageInfo
+from graphql_relay.utils import base64, unbase64
+
+from ..core.enums import OrderDirection
+
+ConnectionArguments = Dict[str, Any]
+
+
+def to_global_cursor(values):
+    if not isinstance(values, list):
+        values = [str(values)]
+    else:
+        values = [str(value) for value in values]
+    return base64("::".join(values))
+
+
+def from_global_cursor(cursor) -> List[str]:
+    values = unbase64(cursor)
+    values = values.split("::")
+    if isinstance(values, list):
+        values = [None if value == "None" else value for value in values]
+        return values
+    if values == "None":
+        values = None
+    return [values]
+
+
+def get_field_value(instance: DjangoModel, field_name: str):
+    """Get field value for given field in filter format 'field__foreign_key_field'."""
+    field_path = field_name.split("__")
+    attr = instance
+    for elem in field_path:
+        attr = getattr(attr, elem)
```

Comment:
no because I don't know where this last elem belong

in simplest case with "__" I could just call
attr = getattr(instance, field_path[1])
attr = getattr(attr, field_path[0])

but I cannot guarantee that this will be the case all the time